index.js

```diff
@@ -1,201 +1,114 @@
-    NOTIFICATION_OPENED_EVENT,
-    IDS_AVAILABLE_EVENT,
-//    SUBSCRIPTION_EVENT,
-//    PERMISSION_EVENT,
-    EMAIL_SUBSCRIPTION_EVENT,
-    IN_APP_MESSAGE_CLICKED_EVENT,
-    // Append new JS string events here
-];
-
-var oneSignalEventEmitter;
-
-var _eventTypeHandler = new Map();
-var _notificationCache = new Map();
-var _listeners = [];
-
-if (RNOneSignal != null) {
-    oneSignalEventEmitter = new NativeEventEmitter(RNOneSignal);
-
-    for(var i = 0; i < _eventBroadcastNames.length; i++) {
-        var eventBroadcastName = _eventBroadcastNames[i];
-        var eventName = _eventNames[i];
-
-        _listeners[eventName] = handleEventBroadcast(eventName, eventBroadcastName)
+    /**
+     * Sets the OneSignal application Id. 1/2 of the initialization process.
+     * @param {string} appId
+     */
+    static setAppId(appId) {
+        RNOneSignal.setAppId(appId);
     }
-}
-
-function handleEventBroadcast(type, broadcast) {
-    return oneSignalEventEmitter.addListener(
-        broadcast, (notification) => {
-            // Check if we have added listener for this type yet
-            // Cache the result first if we have not.
-            var handler = _eventTypeHandler.get(type);
-
-            if (handler) {
-                handler(notification);
-            } else {
-                _notificationCache.set(type, notification);
-            }
-        }
-    );
-}
 
-function checkIfInitialized() {
-    return RNOneSignal != null;
-}
+    /* O B S E R V E R S */
 
-export default class OneSignal {
+    static addPermissionObserver(observer) {
+        if (!checkIfInitialized(RNOneSignal)) return;
+        isValidCallback(observer);
 
-    /**
-     Listen to events of received, opened, ids, subscription, permission, emailSubscription, inAppMessageClicked
-     TODO: We currently have implemented the steps up until connecting the "SUBSCRIPTION_EVENT" and "PERMISSION_EVENT"
-     Currently the getPermissionSubscriptionState is used to get all device information and
-        needs to be broken up into using the native observers to fire these React-Native handlers
-     */
-    static addEventListener(type, handler) {
-        if (!checkIfInitialized()) return;
-
-        invariant(
-            type === NOTIFICATION_RECEIVED_EVENT ||
-            type === NOTIFICATION_OPENED_EVENT ||
-            type === IDS_AVAILABLE_EVENT ||
-//            type === SUBSCRIPTION_EVENT ||
-//            type === PERMISSION_EVENT ||
-            type === EMAIL_SUBSCRIPTION_EVENT ||
-            type === IN_APP_MESSAGE_CLICKED_EVENT,
-            'OneSignal only supports received, opened, ids, emailSubscription, and inAppMessageClicked events'
-        );
-
-        _eventTypeHandler.set(type, handler);
-
-        // Make native request to init notification opened handler
-        if (type === NOTIFICATION_OPENED_EVENT) {
-            RNOneSignal.initNotificationOpenedHandlerParams();
+        if (Platform.OS === 'android') {
```

Comment:
Remove android check, this method is available for iOS as well.

Reply:
Done